roadmap/state_sync_protocol/state-sync-protocol.md

```diff
@@ -0,0 +1,204 @@
+```
+
+**Execution Data (~3 week spork)**
+
+```console
+smnzhu@mbp:~$ gsutil du -sh gs://flow_public_mainnet13_execution_state
+446.14 GiB   gs://flow_public_mainnet13_execution_state
+smnzhu@mbp:~$ gsutil ls gs://flow_public_mainnet13_execution_state | wc -l
+  473590
+```
+
+**Execution State Index (~3 week spork)**
+
+```console
+simon_zhu_dapperlabs_com@observer-001 /var/flow/data/mainnet-13 $ sudo du -sh index
+103G    index
+```
+
+Based on the numbers above, we should require Access nodes to have >43GB of RAM (27GB + current 16GB requirement) and minimum 1.5-2TB of storage space for a 3 month spork. These requirements can likely be decreased if sharding is implemented (see [Future Optimizations](#future-optimizations)).
+
+## Future Optimizations
+
+### Checkpointing
+
+Having a checkpointing mechanism will allow new nodes to catch up more quickly, while also reducing the amount of data Access nodes need to store on disk.
+
+Access nodes can periodically (e.g. at every 10000th block) generate a snapshot of the full state index and store it in their blobstore. They can then prune all of the Execution Data for blocks older than the snapshot, and require nodes that are behind to directly sync the snapshot first before catching up via per-block Execution Data updates.
+
+The root CID of the snapshot must be shared with new nodes somehow so that they can use it to download the snapshot via Bitswap. There are a few ways this can be done:
+
+* The CID can be shared on a broadcast channel using a *Byzantine consistent broadcast* algorithm<sup>[6](#footnotes)</sup>.
+* A new Access node can randomly select a couple of other Access nodes and simply ask them for the CID. If the responses it receives are concordant, it can go ahead and download the data. Once the download is complete, it can generate the state trie for each block and compare its hash to the one found on-chain in the corresponding Execution Result. If it finds an inconsistency, it has all the evidence it needs to submit a slashing challenge against the nodes which provided it with the bad CID.
+
+### Sharding
+
+The Execution Data can be sharded so that each Access node only needs to store and respond to requests for a fraction of the entire dataset.
+
+There are many ways that sharding can be implemented. For example, consistent hashing can be used to determine the assignment of Execution Data to Access nodes, so that when nodes join or leave the network at each epoch, the Execution Data can be remapped according to the consistent hashing scheme. We should use the Random Beacon to determine each Access node's locations around the consistent hashing circle in order to prevent malicious Access nodes from purposely skewing the distribution.
+
+It remains a challenge to ensure that Access nodes keep Execution Data available instead of pruning it and ignoring Bitswap requests. *Data availability proofs* and *proofs of retrievability* may be helpful here:
```

Comment:
Here, I would focus on the ability to horizontally scale the historic state storage within a single access node (instead of sharing across different access nodes)